### PR TITLE
Fixes issue #8032

### PR DESCRIPTION
**Pull request by ChristopheBrun**
- Unit test added -> allows at least to reproduce the bug within the original project
- updated -> PhpManager::updateItem()
- composer.lock optionnaly added into the root .gitignore file

### CODE REVIEW

**Comment by samdark** [comment]
Merged with some adjustments: https://github.com/yiisoft/yii2/commit/82033125c402a43cc17a77e9520054c8e42ad553

Thanks for the fix and especially for test verifying it.

**Comment by qiangxue** [comment]
The change is incorrect - now you can't update the item whose name is not changed.

File: framework/rbac/PhpManager.php [hunk 1]
```diff
@@ -567,15 +567,23 @@ protected function addRule($rule)
...
     /**
+     * @param string $name The name of the item being updated (aka : the 'old' name)
+     * @param Item $item The updated item
+     * @return bool
      * @inheritdoc
      */
     protected function updateItem($name, $item)
     {
+        // You can't overload an item with an existing name
+        if (isset($this->items[$item->name])) {
```
Comment (qiangxue) [inline framework/rbac/PhpManager.php]: If the name is not changed, an exception will be thrown.

Comment (samdark) [inline framework/rbac/PhpManager.php]: Right. It seems I haven't put enough attention. Fixing it.

**Pull request closed without merging**